### DISCUSSION TITLE
Response from CVS - rule|text|element-not-empty

### DISCUSSION THREAD

**Discussion by elerik** [OP-TED/eForms-SDK#597] [Q&A] [answered]
Hi Labs, 

It Would be awesome if this error message will came together with FieldId which is empty :) 
It is pretty hard to figure out from this message 
**"Elements must have content, and the content must be something other than only whitespace."** 
What is wrong. 
I had this for example for this ID **BT-502-Organization-Company**
which is not mandatory and is quite primitive field but in some situations is tested if its not empty. And then it return this very confusing message. 

Thanks.

**Comment by YvesJo** [comment, marked as answer]
Hi,
There is a general rule according to which no leaves should be left empty (an xml element only containing whitespaces is also considered as empty). An XML node, not being a leaf, shall also contain further elements. When a field is not required and not used, then the associated XML element should not be generated.
